internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -677,6 +698,34 @@ func resourceKubernetesClusterNodePoolCreate(d *pluginsdk.ResourceData, meta int
 		return fmt.Errorf("creating %s: %+v", id, err)
 	}
 
+	if subnetID != nil {
+		// Wait for vnet to come back to Succeeded before releasing any locks
+		timeout, _ := ctx.Deadline()
```

Comment:
can we add a sanity check to this:

```suggestion
		timeout, ok := ctx.Deadline()
		if !ok {
			return fmt.Errorf("internal-error: context had no deadline")
		}
```